app/controllers/SdipController.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import config.CSRCache
+import connectors.ApplicationClient
+import forms.SdipForm
+import helpers.NotificationType._
+import security.Roles.{ ActiveUserRole, EditPersonalDetailsRole }
+
+import scala.concurrent.Future
+
+object SdipController extends SdipController(ApplicationClient, CSRCache)
+
+class SdipController(applicationClient: ApplicationClient, cacheClient: CSRCache) extends BaseController(applicationClient, cacheClient) {
```

Comment:
What does SDIP controller do that cannot be in the other controllers? Are we going to have everything SDIP in this controller? Could you give a more precise name? let's say SdipApplicationController or whatever it is.